### ISSUE TITLE
File not found

### ISSUE THREAD

**Issue by varnav** [minio/minio-jenkins#6]
I have equal configurations for standard artifact archive plugin, and minio upload plugin (see screenshot).
Artifacts get archived without problems.
Minio fails with "No such file or directory"

`ERROR: Failed to upload files
java.io.FileNotFoundException: /Volumes/Data/Repo/workspace/openssl_android/deploy/libcrypto.so (No such file or directory)
	at java.io.FileInputStream.open0(Native Method)
	at java.io.FileInputStream.open(FileInputStream.java:195)
	at java.io.FileInputStream.<init>(FileInputStream.java:138)
	at java.io.FileInputStream.<init>(FileInputStream.java:93)
	at org.jenkinsci.plugins.minio.MinioUploader.perform(MinioUploader.java:146)
	at hudson.tasks.BuildStepCompatibilityLayer.perform(BuildStepCompatibilityLayer.java:78)
	at hudson.tasks.BuildStepMonitor$2.perform(BuildStepMonitor.java:32)
	at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:779)
	at hudson.model.AbstractBuild$AbstractBuildExecution.performAllBuildSteps(AbstractBuild.java:720)
	at hudson.model.Build$BuildExecution.post2(Build.java:185)
	at hudson.model.AbstractBuild$AbstractBuildExecution.post(AbstractBuild.java:665)
	at hudson.model.Run.execute(Run.java:1753)
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
	at hudson.model.ResourceController.execute(ResourceController.java:98)
	at hudson.model.Executor.run(Executor.java:404)`
![2](https://cloud.githubusercontent.com/assets/1903613/22290603/e12ba7b8-e312-11e6-9a9d-86a996e9d71a.PNG)

**Comment by varnav** [comment]
Tested with Windows and MacOS slaves, and different paths, including '*.*'
Is it looking for files at slave, not master node?

**Comment by harshavardhana** [comment]
ping @NitishT

**Comment by nitisht** [comment]
Thanks @harshavardhana 
@varnav let me try to reproduce it myself. Will keep you posted.

**Comment by nitisht** [comment]
@varnav the plugin will upload files from the workspace after they are created once the project is built. Looks like the artifact archive plugin deletes the files after upload. Can you confirm this?

**Comment by varnav** [comment]
Artifact archive is not a plugin, it is a core feature of Jenkins. It does not delete files.
I've ran into another problem when testing. I've restarted docker container and it has different credentials now. I've updated credentials in "Configure System", but as I see in the logs, minio plugin still uses old credentials, so I'm getting 'access denied' errors.

**Comment by harshavardhana** [comment]
> I've ran into another problem when testing. I've restarted docker container and it has different credentials now. I've updated credentials in "Configure System", but

You should persist the credentials using environment variables - default docker env is ephemeral 

````
docker run -p 9000:9000 --name minio1 \
  -e "MINIO_ACCESS_KEY=AKIAIOSFODNN7EXAMPLE" \
  -e "MINIO_SECRET_KEY=wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY" \
  -v /mnt/export/minio1:/export \
  -v /mnt/config/minio1:/root/.minio \
  minio/minio server /export
```

**Comment by varnav** [comment]
No, the problem is that minio plugin is not using the credentials that are currently set in "Configure System", it is using the old credentials that were set first time.

**Comment by nitisht** [comment]
Yes, the plugin doesn't know about the Minio server credential every time a new server comes up. To use the same credentials in the plugin, you should persist the credentials when you create the Minio container using environment variables. 

````
docker run -p 9000:9000 --name minio1 \
  -e "MINIO_ACCESS_KEY=AKIAIOSFODNN7EXAMPLE" \
  -e "MINIO_SECRET_KEY=wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY" \
  -v /mnt/export/minio1:/export \
  -v /mnt/config/minio1:/root/.minio \
  minio/minio server /export
````

To change the credentials in the plugin, Go to Manage Jenkins >> Configure System. Then scroll to the Minio upload configuration section & provide the new credentials.

**Comment by varnav** [comment]
So, this means there is no way to change credentials in the plugin? What this setting means then?

![1](https://cloud.githubusercontent.com/assets/1903613/22365737/33d216d2-e48a-11e6-8f0f-c193ce841ebe.PNG)

**Comment by nitisht** [comment]
okay, to clarify 

> To change the credentials in the plugin, Go to Manage Jenkins >> Configure System. Then scroll to the Minio upload configuration section & provide the new credentials.

Does this not work for you?

**Comment by varnav** [comment]
Yes, this does not work for me, as I say above: _I've updated credentials in "Configure System", but as I see in the logs, minio plugin still uses old credentials_

**Comment by nitisht** [comment]
Can you paste the log here?

**Comment by varnav** [comment]
Credential is not equal to one in the settings (see screenshot above)

```
/Volumes/Data/Repo/workspace/openssl_android/deploy/openssl-1.0.2j
ERROR: Minio error
error occured
ErrorResponse(code=AccessDenied, message=Access denied, bucketName=openssl.android, objectName=null, resouce=/openssl.android, requestId=149D924BCF88F5C1, hostId=null)
request={method=HEAD, url=http://10.49.2.22:9000/openssl.android, headers=Host: 10.49.2.22:9000
User-Agent: Minio (amd64; amd64) minio-java/dev
x-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
x-amz-date: 20170127T080632Z
Authorization: AWS4-HMAC-SHA256 Credential=NDCVPVO3QT8LZDSHY3L0/20170127/us-east-1/s3/aws4_request, SignedHeaders=host;x-amz-content-sha256;x-amz-date, Signature=91d4e2413d502a2d1cbd72cf4ed084900bef052a2e5d10d48a6d688e6b400fc7
}
response={code=403, headers=Accept-Ranges: bytes
Server: Minio/RELEASE.2017-01-25T03-14-52Z (linux; amd64)
Vary: Origin
X-Amz-Request-Id: 149D924BCF88F5C1
Date: Fri, 27 Jan 2017 08:05:06 GMT
Content-Type: text/plain; charset=utf-8
OkHttp-Sent-Millis: 1485504392681
OkHttp-Received-Millis: 1485504392684
}

	at io.minio.MinioClient.execute(MinioClient.java:576)
	at io.minio.MinioClient.executeHead(MinioClient.java:646)
	at io.minio.MinioClient.bucketExists(MinioClient.java:1303)
	at org.jenkinsci.plugins.minio.MinioUploader.perform(MinioUploader.java:127)
	at hudson.tasks.BuildStepCompatibilityLayer.perform(BuildStepCompatibilityLayer.java:78)
	at hudson.tasks.BuildStepMonitor$2.perform(BuildStepMonitor.java:32)
	at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:779)
	at hudson.model.AbstractBuild$AbstractBuildExecution.performAllBuildSteps(AbstractBuild.java:720)
	at hudson.model.Build$BuildExecution.post2(Build.java:185)
	at hudson.model.AbstractBuild$AbstractBuildExecution.post(AbstractBuild.java:665)
	at hudson.model.Run.execute(Run.java:1753)
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
	at hudson.model.ResourceController.execute(ResourceController.java:98)
	at hudson.model.Executor.run(Executor.java:404)
Build step 'Upload build artifacts to Minio server' changed build result to UNSTABLE
Finished: UNSTABLE
```

**Comment by nitisht** [comment]
Thanks! Let me go through this and try on my end. 

As for your earlier issue, I tested and was able to upload files after the build process. Can you tell me the environment you are using? Jenkins running in a container?

**Comment by varnav** [comment]
No, Jenkins is not inside a container, it is pure Debian 8 + Jenkins 2.32 from WAR.

**Comment by nitisht** [comment]
Okay, lets clarify the credentials issue and then move to the file not found. I will get back to you after I try this.

**Comment by nitisht** [comment]
Hi @varnav the credentials issue was indeed a bug, its fixed now. Can you please try with the attached .hpi file again. 

You'll need to uninstall the current Minio storage plugin. Then unzip the attached file and install the plugin file (.hpi) via Manage Jenkins >> Manage Plugins >> Advanced >> Upload Plugin. 

I tested again, the file upload is working fine. Please try and let us know how it goes. 
[minio-storage.hpi.zip](https://github.com/minio/minio-jenkins/files/736772/minio-storage.hpi.zip)

**Comment by varnav** [comment]
Settings are fine now, but "file not found" is still there. It sees the file (if I use wildcard, it says exact file name in error message, with correct full path), but can't got further.
I've tested this with different job, with different node (Windows based now), error is the same. Archive artifact option is now turned off, no effect.
Anything I can do to debug this?

**Comment by nitisht** [comment]
@varnav If its okay for you, can we discuss on call and find out the issue? We hang out on slack: https://slack.minio.io . Please join and ping me, so we can decide on a time and see what exactly is causing the issue.

**Comment by nitisht** [comment]
As discussed with @varnav on call today, the plugin works when the build happens on the master. But if the build is done on the slave, files are not uploaded. Will add that feature and upload the .hpi for validation. 

Once validated, will create a new release.

**Comment by nitisht** [comment]
Hey @varnav I have finally been able to add the feature - slave build files should now be uploaded to the Minio server too. 

It would be great if you could try this and let us know if it worked fine for you. Once this is done, we'll make a release. 
[minio-storage.hpi.zip](https://github.com/minio/minio-jenkins/files/798904/minio-storage.hpi.zip)

**Comment by nitisht** [comment]
Closing as it is now possible to upload Jenkins distributed build files Minio via the plugin. Please reopen if there is an issue. 

We released version 0.0.2 with this fix: https://repo.jenkins-ci.org/releases/org/jenkins-ci/plugins/minio-storage/0.0.2/
